### ISSUE TITLE
Calculate gradients and update parameters for CNN+ATT

### ISSUE THREAD

**Issue by pvthuy** [thunlp/NRE#25]
For the code at: https://github.com/thunlp/NRE/blob/master/CNN%2BATT/train.cpp

I can not follow how to calculate gradients and update parameters, maybe from line 193 to line 238.

Could anyone explain, please?

**Comment by Mrlyk423** [comment]
we calculate the loss function from line 114 to 174, and then calculate gradients from 176 to 238.

**Comment by pvthuy** [comment]
Thanks @Mrlyk423 

Can I ask you another question? What is equivalent to the attention matrix att_W at https://github.com/thunlp/NRE/blob/master/PCNN%2BATT/train.cpp?
It seems that you use only the variable `wt = 0.5`.
